var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -52,7 +52,7 @@ class Hdf5(Package):
     variant('cxx', default=True, description='Enable C++ support')
     variant('fortran', default=True, description='Enable Fortran support')
 
-    variant('mpi', default=False, description='Enable MPI support')
+    variant('mpi', default=True, description='Enable MPI support')
```

Comment:
i agree with @adamjstewart that as many things as possible should be `+mpi` by default.